### PR TITLE
[SPARK-48655][SQL] SPJ: Add tests for shuffle skipping for aggregate queries

### PR DESCRIPTION
**Pull request by szehon-ho**
### What changes were proposed in this pull request?
  Add unit test in KeyGroupedPartitionSuite to verify that aggregation can also skip shuffle if key matches partition key

### Why are the changes needed?
  This lacked test coverage

### Does this PR introduce _any_ user-facing change?
  No

### How was this patch tested?
  Ran unit test

### Was this patch authored or co-authored using generative AI tooling?
  No

### CODE REVIEW

**Comment by szehon-ho** [comment]
@sunchao can you take a look?  

Ill next take a look at less group keys than partition keys / incompatible groups, not sure if we can support those.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/KeyGroupedPartitioningSuite.scala [hunk 1]
```diff
@@ -346,6 +352,23 @@ class KeyGroupedPartitioningSuite extends DistributionAndOrderingSuiteBase {
     Column.create("price", FloatType),
     Column.create("time", TimestampType))
 
+  test("group by on partition keys should not introduce additional shuffle") {
```
Comment (sunchao) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/KeyGroupedPartitioningSuite.scala:355]: nit: prefix with JIRA number?

**Review by sunchao — APPROVED** [review]
LGTM

**Comment by sunchao** [comment]
Merged to master, thanks!

**Pull request closed without merging**